core/src/main/java/org/mule/runtime/core/api/config/FeatureContext.java

```diff
@@ -26,7 +27,16 @@
 
   public FeatureContext(MuleVersion artifactMinMuleVersion, String artifactName) {
     this.artifactName = artifactName;
-    this.artifactMinMuleVersion = artifactMinMuleVersion;
+    // Feature flag evaluations must ignore suffixes
+    if (artifactMinMuleVersion != null) {
```

Comment:
better to add this logic to `org.mule.runtime.api.meta.MuleVersion`, add an overloaded method with a flag to indicate to ignore the suffixes

Reply:
I will add a method to the MuleVersion in order to get it without the suffixes, and adjust the feature flags predicates accordingly.